src/entity/Living.php

```diff
@@ -225,6 +226,38 @@ public function setSprinting(bool $value = true) : void{
 			$this->setMovementSpeed($value ? ($moveSpeed * 1.3) : ($moveSpeed / 1.3));
 			$this->moveSpeedAttr->markSynchronized(false); //TODO: reevaluate this hack
 		}
+		$this->recalculateSize();
+	}
+
+	public function isGliding() : bool{
+		return $this->gliding;
+	}
+
+	public function setGliding(bool $value = true) : void{
+		$this->gliding = $value;
+		$this->networkPropertiesDirty = true;
+		$this->recalculateSize();
+	}
+
+	public function isSwimming() : bool{
+		return $this->swimming;
+	}
+
+	public function setSwimming(bool $value = true) : void{
+		$this->swimming = $value;
+		$this->networkPropertiesDirty = true;
+		$this->recalculateSize();
+	}
+
+	public function recalculateSize() : void{
+		$size = $this->getInitialSizeInfo();
+		if($this->isSwimming() || $this->isGliding()){
+			$this->setSize((new EntitySizeInfo($size->getWidth(), $size->getWidth(), $size->getHeight() - $size->getEyeHeight()))->scale($this->getScale()));
+		}elseif($this->isSneaking()){
+			$this->setSize($size->multiply(11 / 12, 1, 11 / 12)->scale($this->getScale()));
```

Comment:
The eye height change here feels wrong. I would expect the distance between eye and top of head to remain the same when crouching, but this would reduce that distance to 11/12.

Reply:
According to [this issue](https://bugs.mojang.com/browse/MC-90598), the correct height for a player would be 1.54 (original height - 0.08).

(Bizarrely, this reduces the distance to top of head to 0.11, while it's 0.15 when not crouching. Still 11/12 is incorrect for eye height - would give 1.485 instead of the expected 1.54.)

Reply:
bedrock is different though

Reply:
Well to my knowledge this is one thing that's the same across both versions (not counting the recent change for slabs).